### PR TITLE
bump version to 4.8.4

### PR DESCRIPTION
**Pull request by ItsGamerik**
this change bumps the version of texstudio to 4.8.2

### CODE REVIEW

**Comment by Zishan-Rahman** [comment]
@ItsGamerik (sorry for the ping) You may know this already, but version 4.8.4 came out a couple of days ago. You may want to update your PR :+1:

**Comment by ItsGamerik** [comment]
thank you for telling me, I updated the manifest

**Comment by Schildkroet** [comment]
The build seems to fail due to a patch

```
Updating files: 100% (10135/10135), done.
Note: switching to '4.8.4'.
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:
  git switch -c <new-branch-name>
Or undo this operation with:
  git switch -
Turn off this advice by setting config variable advice.detachedHead to false
HEAD is now at 6af6660 Cwls (#3814)
FB: Running: git lfs checkout
Checking out LFS objects:   0% (0/1), 0 B | 0 B/s
Checking out LFS objects: 100% (1/1), 0 B | 0 B/s, done.
FB: Running: git rev-parse --verify --quiet 4.8.4:.gitmodules
Applying patch 0001-Ensure-the-initial-commands-are-properly-set-when-we.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-3/build/0001-Ensure-the-initial-commands-are-properly-set-when-we.patch
patching file src/buildmanager.cpp
command timed out: 3600 seconds without output running [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'x86_64', b'--bundle-sources', b'--mirror-screenshots-url=https://dl.flathub.org/media', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-x86_64-3/../downloads', b'--default-branch', b'test', b'--subject', b'Update poppler, languagetool and texstudio modules (3e794531)', b'builddir', b'org.texstudio.TeXstudio.yaml', b'--disable-download'], attempting to kill
process killed by signal 9
program finished with exit code -1
elapsedTime=3669.881111
Reversed (or previously applied) patch detected!  Assume -R? [n] 
```

**Comment by iaacornus** [comment]
> The build seems to fail due to a patch
> 
> ```
> Updating files: 100% (10135/10135), done.
> Note: switching to '4.8.4'.
> You are in 'detached HEAD' state. You can look around, make experimental
> changes and commit them, and you can discard any commits you make in this
> state without impacting any branches by switching back to a branch.
> If you want to create a new branch to retain commits you create, you may
> do so (now or later) by using -c with the switch command. Example:
>   git switch -c <new-branch-name>
> Or undo this operation with:
>   git switch -
> Turn off this advice by setting config variable advice.detachedHead to false
> HEAD is now at 6af6660 Cwls (#3814)
> FB: Running: git lfs checkout
> Checking out LFS objects:   0% (0/1), 0 B | 0 B/s
> Checking out LFS objects: 100% (1/1), 0 B | 0 B/s, done.
> FB: Running: git rev-parse --verify --quiet 4.8.4:.gitmodules
> Applying patch 0001-Ensure-the-initial-commands-are-properly-set-when-we.patch
> FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-3/build/0001-Ensure-the-initial-commands-are-properly-set-when-we.patch
> patching file src/buildmanager.cpp
> command timed out: 3600 seconds without output running [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'x86_64', b'--bundle-sources', b'--mirror-screenshots-url=https://dl.flathub.org/media', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-x86_64-3/../downloads', b'--default-branch', b'test', b'--subject', b'Update poppler, languagetool and texstudio modules (3e794531)', b'builddir', b'org.texstudio.TeXstudio.yaml', b'--disable-download'], attempting to kill
> process killed by signal 9
> program finished with exit code -1
> elapsedTime=3669.881111
> Reversed (or previously applied) patch detected!  Assume -R? [n] 
> ```

any idea what caused it?

**Comment by Schildkroet** [comment]
It tries to patch a file, but times out due to no user reaction. But this problem seems to be solved.
```
patching file src/buildmanager.cpp
command timed out: 3600 seconds without output running [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'x86_64', b'--bundle-sources', b'--mirror-screenshots-url=https://dl.flathub.org/media', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-x86_64-3/../downloads', b'--default-branch', b'test', b'--subject', b'Update poppler, languagetool and texstudio modules (3e794531)', b'builddir', b'org.texstudio.TeXstudio.yaml', b'--disable-download'], attempting to kill
process killed by signal 9
```

Now it reports a validation error
```
flatpak run --command=flatpak-builder-lint org.flatpak.Builder --exceptions repo repo
 in dir /srv/buildbot/worker/build-x86_64-4/build (timeout 1200 secs)
 watching logfiles {}
 argv: b'flatpak run --command=flatpak-builder-lint org.flatpak.Builder --exceptions repo repo'
 using PTY: False
{
    "errors": [
        "appstream-failed-validation"
    ],
    "warnings": [
        "appstream-screenshot-missing-caption"
    ],
    "appstream": [
        "W: org.texstudio.TeXstudio:64: tag-empty ul/li"
    ],
    "info": [
        "appstream-failed-validation: Metainfo file /tmp/tmpzrqon74s/metainfo/org.texstudio.TeXstudio.metainfo.xml has failed validation. Please see the errors in appstream block",
        "appstream-screenshot-missing-caption: One or more screenshots are missing captions in the Metainfo file"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
program finished with exit code 1
elapsedTime=12.983282
```

**Comment by Schildkroet** [comment]
The error seems to differ between different PRs. Since i can't see a date, i don't know that the current problem is, or if there are several issues

**Comment by Schildkroet** [comment]
Who is the maintainer of this repo? There seems to be no interaction anymore...

**Comment by ItsGamerik** [comment]
The reference to the patch in the flatpak manifest already has been removed, so that should not be an issue anymore. The flatpak also build successfully [after this change](https://github.com/flathub/org.texstudio.TeXstudio/pull/213#issuecomment-2375120624), so the last remaining issue is the metainfo problem.
I suspect that the shell script used to generate the metainfo is the culprit; I will try to fix it and update the PR to then hopefully get it merged :)

**Comment by ItsGamerik** [comment]
So I think the error is related to the fact that the changelog for texstudio version 4.8.4 is empty. 
The script tries to parse the content of [CHANGELOG.md](https://github.com/texstudio-org/texstudio/blob/4.8.4/utilities/manual/source/CHANGELOG.md), but the list is empty so it generates xml which is not accepted by the flathub xml linter. 
This is changed in the following versions, we will see what happens when we bump the version again.

**Comment by iaacornus** [comment]
so, how do we merge this to get the 4.8.4 version of texstudio in stable?

**Comment by iaacornus** [comment]
who else has access to the repository? Alexander Wilms' acc seems to be deactivated or deleted ...

**Comment by ItsGamerik** [comment]
> so, how do we merge this to get the 4.8.4 version of texstudio in stable?

I think this PR for 4.8.4 should be good to go with the changes I made to the bash script.


> who else has access to the repository? Alexander Wilms' acc seems to be deactivated or deleted ...

@jakobjakobson13 responded to the original Issue (#212), so maybe he can merge this? Last update was quite a while ago though...

File: 0001-Ensure-the-initial-commands-are-properly-set-when-we.patch.old
Comment (hfiguiere) [inline 0001-Ensure-the-initial-commands-are-properly-set-when-we.patch.old:1]: remove this file since it's no longer needed.

**Comment by bbhtt** [comment]
Do you want to take care of the poppler and languagetool updates as well?

**Comment by ItsGamerik** [comment]
I guess that would just require a rebase by the bot on #219 no?

**Comment by bbhtt** [comment]
Oh yea I thought they were failing for other reasons but looks like the latest update PR passes.

**Pull request merged**